tests/measurements/test_shots.py

```diff
@@ -250,3 +250,35 @@ def test_has_partitioned_shots(self, shots, expected):
     def test_num_copies(self, shots, expected):
         """Tests the num_copies property."""
         assert Shots(shots).num_copies == expected
+
+    def test_shot_mul(self):
+        """Test the __mul__ method for multiplying a number by a shot object."""
+        sh1, sh2, sh3 = Shots(100), Shots((100, 100)), Shots(5)
+        sh5 = Shots()
+        scaled_sh1 = sh1 * 2
+        scaled_sh2 = sh2 * 2
+        scaled_sh3 = sh3 * 0.5
+        scaled_sh4 = 2 * sh2
+        scaled_sh5 = sh5 * 8
+
+        assert scaled_sh1.total_shots == 200
+        assert scaled_sh2.total_shots == 400
+        assert scaled_sh2.shot_vector[0].shots == 200
+        assert scaled_sh2.shot_vector[0].copies == 2
+        assert scaled_sh3.total_shots == 2
+        assert scaled_sh4.total_shots == 400
+        assert scaled_sh4.shot_vector[0].shots == 200
+        assert scaled_sh4.shot_vector[0].copies == 2
+        assert scaled_sh5 == sh5
+
+    def test_invalid_scalar_type(self):
+        """Test that __mul__ raises a TypeError for an invalid scalar type."""
+        shots = Shots(100)
+        # pylint: disable=pointless-statement
+        with pytest.raises(TypeError, match="can't multiply sequence by non-int of type 'Shots'"):
+            shots * "invalid scalar type"  # pylint: disable=unused-variable
+        # pylint: enable=pointless-statement
```

Comment:
```suggestion
            _ = shots * "invalid scalar type"
```
This way you don't need to disable the pylint warnings

Reply:
The reason why you initially got the `pointless-statement` pylint error is because `Shots.__mul__` returns a `Shots` object, but you didn't assign it to anything. By making the change I suggested, the returned object is assigned to `_`, which is conventionally used for assignments you don't care about. So, pylint won't throw a warning.

Reply:
Actually i have even tried that possibility, and if i remove the pylint disable statements as you have suggested above:
You can see again pylint is throwing warning and eventually the same will be detected here also.


```
def test_invalid_scalar_type(self):
        """Test that __mul__ raises a TypeError for an invalid scalar type."""
        shots = Shots(100)
        with pytest.raises(TypeError, match="Can't multiply Shots with non-integer or float type."):
            invalid_shot=shots * "invalid scalar type" 
```


![image](https://github.com/PennyLaneAI/pennylane/assets/79191029/03cf22bb-5f6e-44aa-9051-129d95d92a15)

Reply:
Ah I see. This is because you're assigning `shots * "invalid scalar type" to a named variable called `invalid_shots`. To get rid of the warning, you need to assign it to an unnamed variable, which is why I suggested using `_ = shots * invalid scalar type". Let me know if that doesn't work. It's usually best to avoid disabling pylint warnings unless necessary 😄